python/src/scipp/compat/mantid.py

```diff
@@ -82,25 +82,49 @@ def make_detector_info(ws):
                                         labels={'spectrum': spectrum}))
 
 
+def md_dimension(mantid_dim, index):
+    patterns = ["^q.*{0}$".format(coord) for coord in ['x', 'y', 'z']]
+    q_dims = [sc.Dim.Qx, sc.Dim.Qy, sc.Dim.Qz]
+    pattern_result = zip(patterns, q_dims)
+    if mantid_dim.getMDFrame().isQ():
+        for pattern, result in pattern_result:
+            if re.search(pattern, mantid_dim.name, re.IGNORECASE):
+                return result
+    return [sc.Dim.X, sc.Dim.Y, sc.Dim.Z][index]
```

Comment:
So Mantid supports only Q and real space? What about a forth dimension, such as `DeltaE`, or `Temperature`?

Reply:
As commented on in PR title, I'm not yet considering this. The biggest hurdle is to identify this from the input, but I'll attempt to improve.